### ISSUE TITLE
use user-error to signal errors

### ISSUE THREAD

**Issue by wbolster** [sshaw/git-link#45]
first, thanks for this package!

that said, please use `(user-error ...)` instead or `(message ...)` to signal errors, so that i can catch them and do something sensible.

this would allow me to write a function that would first try `(git-link-commit)`, falling back to `(git-link)` in case that fails, which would basically "do what i mean" in 99% of my use cases.

let me know whether i should submit a pr!

**Comment by sshaw** [comment]
Hi, I only call `error` [in `git-link-homepage`](https://github.com/sshaw/git-link/blob/master/git-link.el#L470). `messages` is most everywhere. 

Can you provide more info? Thanks!

**Comment by wbolster** [comment]
yeah, my suggestion is to use `(user-error)` instead of `(message)` when something goes wrong. right now you use `(message)` which does not trigger an error that can be caught.

i want to be able to do something like this:

``` elisp
    (condition-case nil
        (git-link-commit)
      (user-error
       (git-link)))
```

but it seems right now i cannot.

**Comment by sshaw** [comment]
Got it, makes sense. I'll have to look into user-error. 

You may also want to checkout the PRs and issues. There's talk of making a `git-link-dwim` function or maybe even changing `git-link` to fallback to `git-link-commit`. 

I think changing these functions to signal errors will require releasing a 1.0 version.

**Comment by wbolster** [comment]
hey @sshaw 👋🏼, long time no see 🙃

i recently noticed that `git-link` doesn't report errors correctly, looked at the issue tracker, and to my own surprise, it was _me_ who first opened this issue… back in 2017! 🤯

let's get it fixed once and for all. i just opened #154, so please take a look!
